psi4/src/psi4/libmints/writer.cc

```diff
@@ -563,21 +610,23 @@ void FCHKWriter::write(const std::string &filename) {
     write_matrix("Contraction coefficients", coefficients);
     write_matrix("Coordinates of each shell", shell_coords);
     write_number("Total Energy", wavefunction_->energy());
-    // write_matrix("Alpha Orbital Energies", wavefunction_->epsilon_a_subset("AO"));
-    write_matrix(wavefunction_->epsilon_a()->name().c_str(), wavefunction_->epsilon_a_subset("AO"));
-    // write_matrix("Alpha MO coefficients", reorderedCa);
-    write_matrix(wavefunction_->Ca()->name().c_str(), reorderedCa);
-    // write_matrix("Beta Orbital Energies", wavefunction_->epsilon_b_subset("AO"));
-    write_matrix(wavefunction_->epsilon_b()->name().c_str(), wavefunction_->epsilon_b_subset("AO"));
-    // write_matrix("Beta MO coefficients", reorderedCb);
-    write_matrix(wavefunction_->Cb()->name().c_str(), reorderedCb);
-    auto *label = new char[256];
-    std::string type = name == "DFT" ? "SCF" : name;
-    sprintf(label, "Total %s Density", type.c_str());
-    write_sym_matrix(label, reorderedDt);
-    sprintf(label, "Spin %s Density", type.c_str());
-    write_sym_matrix(label, reorderedDs);
-    delete[] label;
+
+    // This is the format expected in the formatted checkpoint file
+    write_matrix("Alpha Orbital Energies", wavefunction_->epsilon_a_subset("AO"));
+    write_matrix("Alpha MO coefficients", reorderedCa);
+    write_sym_matrix("Total SCF Density", reorderedDt);
+    // In theory, the correlated density should be printed out with a
+    // legend depending on the method (MP2, MP3, MP4, CI, CC) but this
+    // is probably the most common anyhow
+    if (pHF) write_sym_matrix("Total CC Density", reorderedDPHFt);
```

Comment:
This isn't just the CC density? Could be many theories.

Reply:
[This](http://wild.life.nctu.edu.tw/~jsyu/compchem/g09/g09ur/f_formchk.htm) looks like a description of the fchk format that might have been in an earlier version of the Gaussian manual. I can't find similar in current docs.

The headers for `Total <mtd> Density` are so general (SCF, MP2, CI, CC) that CC may be reasonable. Or, since this is called from py, can pull `name` off wfn, regex it into SCF/MP2/CI/CC and pass that label into the fchk writer. In any case, the 2nd line of the fchk, `Type, Method, Basis`, provide a place to put the proper, e.g., `CISD` method.

Other than tidying up the name, lgtm! oh, and should probably check that gdma is content with the changes. Thanks for fixing this up.

Reply:
As discussed elsewhere, this is just an aesthetic issue. Since the checkpoint never includes other than the SCF and one unrelaxed post-HF density, it doesn't really matter how the post-HF density is named - as long as the name doesn't violate the standard. Since >99% of use cases are likely CC, I don't see why this would need to be changed here.